hack/test-cmd.sh

```diff
@@ -202,6 +203,7 @@ kube::util::wait_for_url "http://127.0.0.1:${API_PORT}/healthz" "apiserver"
 kube::log::status "Starting controller-manager"
 "${KUBE_OUTPUT_HOSTBIN}/kube-controller-manager" \
   --port="${CTLRMGR_PORT}" \
+  --kube-api-content-type="${KUBE_TEST_API_TYPE-}" \
```

Comment:
nit: we should probably start all other components in the same mode too (scheduler, kubelet, kube-proxy). But we can do it in a separate PR probably.